powerstore/resource_snapshotrule.go

```diff
@@ -0,0 +1,266 @@
+package powerstore
+
+import (
+	"context"
+	"log"
+	"terraform-provider-powerstore/models"
+
+	"github.com/dell/gopowerstore"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type resourceSnapshotRuleType struct{}
+
+// GetSchema returns the schema for snapshotrule resource.
+func (r resourceSnapshotRuleType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:                types.StringType,
+				Computed:            true,
+				Description:         "The ID of the snapshot rule.",
+				MarkdownDescription: "The ID of the snapshot rule.",
+			},
+			"name": {
+				Type:                types.StringType,
+				Required:            true,
+				Description:         "The name of the snapshot rule.",
+				MarkdownDescription: "The name of the snapshot rule.",
+			},
+			"interval": {
```

Comment:
Is this param mutually exclusive with time_of_day? If yes, please specify in the description.